power-management/rapl/powercap/runtest.sh

```diff
@@ -0,0 +1,201 @@
+        (( e1 += e_cpu ))
+    done
+
+    sleep $time
+
+    typeset e2=0
+    for cpu in $rapl; do
+        typeset e_cpu=$(cat "$rapl_path/$cpu/energy_uj"*)
+        (( e2 += e_cpu ))
+    done
+
+    typeset e=$(( (e2 - e1) / time ))
+    echo $e
+}
+
+# === power-management/rapl/powercap/load1.sh =================================
+function load1
+{
+    typeset sleeptime=${1:-"1"}
+
+    # do some load
+    dd if=/dev/zero > /dev/null 2>&1 &
+    typeset loadpid=$!
+
+    sleep $sleeptime
+    # wait a bit longer, than length of test
+    sleep 10
+
+    # kill the load
+    kill $loadpid
+}
+
+# === power-management/rapl/powercap/load_m.sh ================================
+function loadm
+{
+    typeset -i timeout=$1
+    typeset -i nthreads=$2
+    typeset -i i
+    for (( i = 0; i < $nthreads; i++ )); do
+        (load1 $timeout) > /dev/null 2>&1 &
+    done
+}
+
+# === power-management/rapl/powercap/test_capping.sh ==========================
+function test_capping
+{
+    # measurement time should be bigger than
+    # cat /sys/devices/virtual/powercap/intel-rapl/intel-rapl\:0/constraint_0_time_window_us
+    # which is 1 sec by default (at least on systems I have seen so far).
+    typeset measurement_time=30
+    typeset load_time=$(( measurement_time + 20 ))
+
+    # 1. no load, no cap
+    typeset e_no_no=$(get_energy $measurement_time)
+    rlLog "Power consumtion on uncapped system with no load: $e_no_no microwatts"
+
+    # 2. load, no cap
+    typeset cpus=$(lscpu | grep '^CPU(s):' | \
+                   sed 's/^CPU(s):[^0-9]*\([0-9]*\)/\1/')
+    loadm $load_time $cpus
+    typeset e_full_no=$(get_energy $measurement_time)
+    rlLog "Power consumtion on uncapped system with full load: $e_full_no microwatts"
+
+    # 3. cap to average between full load and no load
+    typeset e_mid=$(( (e_no_no + e_full_no) / 2 ))
+    rlLog "Going to cap the system to: $e_mid microwatts"
+
+    typeset old_cap=$(get_cap)
+    set_cap $e_mid
+    loadm $load_time $cpus
+    typeset e_full_cap=$(get_energy $measurement_time)
+    rlLog "Power consumtion on capped system with full load: $e_full_cap microwatts"
+
+    #
+    # measured_max is maximal measured value which will not cause fail
+    # measured_min is minimal measured value which will not cause fail
+    # *11/10 - add 10% in bash which works in integers only - we're
+    # using microwatts, so numbers are milions to hundreds of milions,
+    # so tehre is no problem with rounding error
+    # *9/10 - substract 10%
+    #
+    typeset measured_max=$(( e_mid * 11 / 10 ))
+    typeset measured_min=$(( e_mid *  9 / 10 ))
+
+    rlLog "capping back to $old_cap"
+    set_cap $old_cap
+
+    if (( e_full_cap > measured_max || e_full_cap < measured_min )); then
+        rlLog "FAIL: measured energy consumption doesn't match capped value"
+        return 1
+    else
+        rlLog "PASS: Test passed"
+    fi
+
+    # All my tests can write warnings (like non met depedencies) to this
```

Comment:
If the dependencies aren't met, can we instead ABORT or SKIP based on what it is? We don't want to tell developers their patches are wrong because a test dependency wasn't met.

Reply:
Have added a function `is_intel()` to startup(). If the CPU is not intel, will abort the case.